hopli/src/environment_config.rs

```diff
@@ -1,64 +1,49 @@
 
-impl fmt::Display for EnvironmentType {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            Self::Production => write!(f, "production"),
-            Self::Staging => write!(f, "staging"),
-            Self::Development => write!(f, "development"),
-            Self::Local => write!(f, "local"),
-        }
-    }
-}
+use crate::utils::HelperErrors;
+use chain_api::{DefaultHttpPostRequestor, JsonRpcClient};
+use chain_rpc::{client::SimpleJsonRpcRetryPolicy, errors::RpcError, rpc::RpcOperationsConfig};
+use clap::Parser;
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware},
+    providers::{Middleware, Provider},
+    signers::{LocalWallet, Signer, Wallet},
+};
+use hopr_crypto_types::keypairs::ChainKeypair;
+use hopr_crypto_types::keypairs::Keypair;
+use hopr_lib::{EnvironmentType, NetworkContractAddresses};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
 
-/// Contract addresses and configuration associated with a environment
-#[derive(Debug, Clone, Serialize, Deserialize)]
+// replace NetworkConfig with ProtocolConfig
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(deny_unknown_fields)]
 pub struct NetworkDetail {
-    /// block number from which the indexer starts
+    /// block number to start the indexer from
     pub indexer_start_block_number: u32,
     /// Type of environment
+    #[serde_as(as = "DisplayFromStr")]
     pub environment_type: EnvironmentType,
-    /// Contract addresses
-    pub addresses: Addresses,
-}
-
-/// Contract addresses (directly from deployment logic)
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct Addresses {
-    /// address of contract that manages authorization to access the Hopr network
-    pub network_registry: String,
-    /// address of contract that maps to the requirements that need to be fulfilled
-    /// in order to access the network, upgradeable
-    pub network_registry_proxy: String,
-    /// HoprChannels contract address, implementation of mixnet incentives
-    pub channels: String,
-    /// Hopr token contract address
-    pub token: String,
-    /// contract address of Safe capability module implementation
-    pub module_implementation: String,
-    /// address of contract that maps between Safe instances and node addresses
-    pub node_safe_registry: String,
-    /// address of contract that allows Hopr Association to dictate price per packet in Hopr
-    pub ticket_price_oracle: String,
-    /// address of contract that manages transport announcements in the hopr network
-    pub announcements: String,
-    /// factory contract to produce Safe instances
-    pub node_stake_v2_factory: String,
+    /// contract addresses used by the network
```

Comment:
It'd be ideal to have docs comments start with a capital letter. You can generate the docs with `cargo doc --no-deps` in the workspace root and browse through the generated content in the browser.